src/wp-includes/option.php

```diff
@@ -277,6 +277,22 @@ function form_option( $option ) {
 function wp_load_alloptions( $force_cache = false ) {
 	global $wpdb;
 
+	/**
+	 * Filters the array of alloptions before it is populated.
+	 *
+	 * Returning a non-null value from the filter will effectively short circuit
+	 * wp_load_alloptions(), returning that value instead.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param null|array $alloptions  An array of alloptions.
+	 * @param bool       $force_cache Whether to force an update of the local cache from the persistent cache. Default false.
+	 */
+	$alloptions = apply_filters( 'pre_wp_load_alloptions', null, $force_cache );
+	if ( null !== $alloptions ) {
```

Comment:
Hmm, should this be `is_array( $alloptions )` to ensure the function's return annotation is still accurate?

Otherwise, `php
add_filter( 'pre_wp_load_alloptions', '__return_false' );` will short-circuit this function and return `false`.

If, for some reason, extenders wanted an empty value to return, they could still use `add_filter( 'pre_wp_load_alloptions', '__return_empty_array' );`

It seems reasonable that we would otherwise prevent non-array values from being returned here.

Reply:
> Hmm, should this be `if ( is_array( $alloptions ) )` to ensure the function's return annotation is still accurate?

I think this would be a good change. The hooks docblock would need an update to reflect this.